### ISSUE TITLE
Enable debugging Edge

### ISSUE THREAD

**Issue by alexcope** [Microsoft/IEDiagnosticsAdapter#48]
The IEDiagnosticsAdapter should also be able to connect to Edge.

**Comment by RReverser** [comment]
Isn't Edge sharing same protocol?

I mean this tool recognizes Edge tabs perfectly, I can see their titles, URLs etc., and just getting this when actually attempting to connect:

> DEBUG MESSAGE: Attempting to attach to 64 bit tab
Connection rejected for: /devtools/page/00C668E2-76B5-4EB4-B89F-5C6314DD1F37
[2015-08-09 20:00:09] [error] Handshake ended with HTTP error: 400

So looks like the problem is just lack of support of 64-bit tabs, but not of Edge in general.

**Comment by andysterland** [comment]
tl;dr: Yes, the protocol is about the same but Edge is a different kind of app so the ways which you find it and load into it are different.

They API surface is roughly the same though there are the odd breaking changes between Edge and IE11.

For the adapter we already ship 64bit versions of the component (proxy.dll) that loads in the tab, as you can still have 64bit tabs in IE11 it's just much, much, less common. 

One example of a significant change is just the permission model as Edge content processes run in an app container which is an even more restricted than LowIL etc. and an ACL has to be added to the DLL to enable it to be loaded into the Edge process. If I recall we also need to add a manifest entry to the exe to allow it to enumerate 'modern' windows. Basically one of the ways that the adapter finds the web pages to debug is by enumerating all the running windows on the desktop looking for ones that match the class name of "Internet Explorer_Server". By default modern windows (ICoreWindow) aren't returned via those window enumeration APIs. But, that permission can be granted via a manifest entry. So, there is some work to be done.

**Comment by auchenberg** [comment]
Edge Debugging has been added in a fork of this adaptor. Any future work related to Microsoft Edge will take place here, https://github.com/Microsoft/edge-diagnostics-adaptor
